src/platform/backends/qemu/qemu_virtual_machine_factory.cpp

```diff
@@ -205,8 +206,9 @@ mp::VirtualMachine::UPtr mp::QemuVirtualMachineFactory::create_virtual_machine(c
     set_ip_forward();
     set_nat_iptables();
 
-    return std::make_unique<mp::QemuVirtualMachine>(desc, ip_pool.obtain_ip_for(desc.vm_name), tap_device_name,
-                                                    generate_mac_address(), monitor);
+    return std::make_unique<mp::QemuVirtualMachine>(
+        desc, ip_pool.check_ip_for(desc.vm_name).value_or(mp::IPAddress{0xFFFFFFFF}), tap_device_name,
```

Comment:
QemuVirtualMachine should also take a std::experimental::optional<IPAddress> which then would allow you to remove:
.value_or(mp::IPAddress{0xFFFFFFFF})